### ISSUE TITLE
An RMI Against An Image ID That Fails Causes Image To Be Untagged

### ISSUE THREAD

**Issue by amcrn** [docker/docker#20469]
`docker version`: Docker version 1.9.1, build a34a1d5
`docker info`:

```
Containers: 4
Images: 53
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 61
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 2
Total Memory: 5.83 GiB
Name: REDACTED
ID: REDACTED
WARNING: No swap limit support
```

`uname -a`:

```
Linux REDACTED #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux`
```

Environment details: VM.

How reproducible: 100% of the time.

Steps to Reproduce:

1) Run a container with an image using `-d`
2) `docker rmi image_name:tag`

The above will complain that there's an active container running. Run `docker ps` and `docker images` and note how the image you used in Step 1 is still labeled correctly.

3) `docker rmi <id>`, where `<id>` is the id of the image you used in the prior step.

The above will complain that there's an active container running. However, now when you run `docker ps` and `docker images`, the image is now untagged.

Example:

```
stack@myvm:~$ docker ps
CONTAINER ID        IMAGE                                        COMMAND                  CREATED             STATUS              PORTS               NAMES
2a45e073cc07        myorg/myproj/logstash:2.1.0-1                "/docker-entrypoint.s"   10 days ago         Up 5 days                               logstash

stack@myvm:~$ docker rmi myorg/myproj/logstash:2.1.0-1
Error response from daemon: conflict: unable to remove repository reference "myorg/myproj/logstash:2.1.0-1" (must force) - container 2a45e073cc07 is using its referenced image 805530fa1bc0
Error: failed to remove images: [myorg/myproj/logstash:2.1.0-1]

stack@myvm:~$ docker images
REPOSITORY                           TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
myorg/myproj/logstash                2.1.0-1             805530fa1bc0        12 weeks ago        444.7 MB


stack@myvm:~$ docker rmi 805530fa1bc0
Error response from daemon: conflict: unable to delete 805530fa1bc0 (cannot be forced) - image is being used by running container 2a45e073cc07
Error: failed to remove images: [805530fa1bc0]


stack@myvm:~$ docker ps
CONTAINER ID        IMAGE                                       COMMAND                  CREATED             STATUS              PORTS               NAMES
2a45e073cc07        805530fa1bc0                                "/docker-entrypoint.s"   10 days ago         Up 5 days                               logstash

stack@myvm:~$ docker images
REPOSITORY                           TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
<none>                               <none>              805530fa1bc0        12 weeks ago        444.7 MB
```

Expected Result: A failed `rmi` against an image id should behave the same as an `rmi` against a name+tag in the sense that it should not mutate anything.

**Comment by aaronlehmann** [comment]
Thanks for the report. This has been fixed in Docker 1.10 - see #18873 and https://github.com/docker/docker/pull/19266.

**Comment by amcrn** [comment]
I'm not sure how my searching sucked so bad that I didn't find those. Apologies. Appreciate the quick response!
